presto-orc/src/main/java/com/facebook/presto/orc/writer/CompressionBufferPool.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.writer;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import static io.airlift.slice.SizeOf.sizeOf;
+
+public interface CompressionBufferPool
+{
+    byte[] checkOut(int length);
+
+    void checkIn(byte[] buffer);
+
+    long getRetainedBytes();
+
+    class LastUsedCompressionBufferPool
+            implements CompressionBufferPool
+    {
+        private static final int INSTANCE_SIZE = ClassLayout.parseClass(LastUsedCompressionBufferPool.class).instanceSize();
+        private byte[] lastUsed;
+
+        @Override
+        public byte[] checkOut(int length)
+        {
+            if (lastUsed == null || lastUsed.length < length) {
+                return new byte[length];
```

Comment:
Do you want to nullify lastUsed?

Reply:
They will be taken care of when they are checked in, but we can nullify here as well.